### PR TITLE
Miscellaneous fixes.

### PR DESCRIPTION
**Pull request by maciejkula**
A bunch of fixes; I'll  put comments next to my changes.

### CODE REVIEW

File: spotlight/factorization/explicit.py [hunk 1]
```diff
@@ -230,7 +230,7 @@ def fit(self, interactions, verbose=False):
 
                 self._optimizer.zero_grad()
 
-                loss = self._loss_func(minibatch.ratings, predictions, minibatch.weights)
+                loss = self._loss_func(minibatch.ratings, predictions, weights=minibatch.weights)
```
Comment (maciejkula) [inline spotlight/factorization/explicit.py:233]: Using a keyword arg here: don't want to rely on the order.

File: spotlight/losses.py [hunk 2]
```diff
@@ -242,7 +242,7 @@ def poisson_loss(observed_ratings, predicted_ratings, weights=None):
 
     assert_no_grad(observed_ratings)
 
-    loss = (predicted_ratings - observed_ratings) * torch.log(predicted_ratings)
+    loss = predicted_ratings - observed_ratings * torch.log(predicted_ratings)
```
Comment (maciejkula) [inline spotlight/losses.py:245]: Was your change intentional? I'm not entirely sure if what it was was right, but this change certainly made the loss worse in the unit test.

File: tests/factorization/test_explicit.py [hunk 3]
```diff
@@ -6,51 +6,25 @@
...
 from spotlight.factorization.explicit import ExplicitFactorizationModel
 
-import torch
-from torch.autograd import Variable
 
 RANDOM_STATE = np.random.RandomState(42)
 CUDA = bool(os.environ.get('SPOTLIGHT_CUDA', False))
 
 
-def test_regression_loss():
```
Comment (maciejkula) [inline tests/factorization/test_explicit.py:19]: Moved these to their own file.

File: tests/factorization/test_explicit.py [hunk 4]
```diff
@@ -6,51 +6,25 @@
...
-    assert out.data.numpy().sum().item() == 0
-
-
 @pytest.mark.parametrize('weight_factor, loss, expected', [
-    (1, 'regression', 0.5),
+    (1, 'regression', 1.0),
     (0, 'regression', 1.0),
     (1, 'poisson', 1.0),
-    (0, 'poisson', 2.0)
+    (0, 'poisson', 1.0)
```
Comment (maciejkula) [inline tests/factorization/test_explicit.py:20]: Reverting the deterioration in the poisson loss RMSE>

File: tests/factorization/test_explicit.py [hunk 5]
```diff
@@ -6,51 +6,25 @@
...
 ])
 def test_model_fitting(weight_factor, loss, expected):
 
     interactions = movielens.get_movielens_dataset('100K')
 
     # Add weights
-    interactions.weights = weight_factor * np.ones((100000,))
+    interactions.weights = np.repeat(weight_factor, len(interactions))
```
Comment (maciejkula) [inline tests/factorization/test_explicit.py:27]: Using `len(interactions)` gets rid of the magic number.

File: tests/factorization/test_explicit.py [hunk 6]
```diff
@@ -59,14 +33,19 @@ def test_model_fitting(weight_factor, loss, expected):
...
-                                       l2=1e-5,
+                                       l2=1e-6,
                                        use_cuda=CUDA)
 
     model.fit(train)
 
     rmse = rmse_score(model, test)
 
-    assert rmse > expected
+    if weight_factor == 0:
```
Comment (maciejkula) [inline tests/factorization/test_explicit.py:43]: The same thing you did in the implicit tests.

File: spotlight/losses.py (hunk 2 above)
Comment (lindauruchurtu) [reply spotlight/losses.py:245]: This was definitely not intentional.

File: tests/factorization/test_explicit.py (hunk 3 above)
Comment (lindauruchurtu) [reply tests/factorization/test_explicit.py:19]: 👍

File: tests/factorization/test_explicit.py (hunk 5 above)
Comment (lindauruchurtu) [reply tests/factorization/test_explicit.py:27]: Good one.

File: spotlight/factorization/explicit.py (hunk 1 above)
Comment (lindauruchurtu) [reply spotlight/factorization/explicit.py:233]: That makes sense

File: tests/factorization/test_explicit.py (hunk 4 above)
Comment (lindauruchurtu) [reply tests/factorization/test_explicit.py:20]: 👍

File: tests/factorization/test_explicit.py (hunk 6 above)
Comment (lindauruchurtu) [reply tests/factorization/test_explicit.py:43]: I missed this one so thanks for adding.

**Review by lindauruchurtu — APPROVED** [review]
👍
